packages/dd-trace/src/appsec/index.js

```diff
@@ -90,18 +102,21 @@ function incomingHttpStartTranslator ({ req, res, abortController }) {
     payload[addresses.HTTP_CLIENT_IP] = clientIp
   }
 
+  if (sampleRequest(config.appsec.apiSecurity)) {
+    payload[addresses.WAF_CONTEXT_PROCESSOR] = { 'extract-schema': true }
+  }
+
   const actions = waf.run(payload, req)
 
   handleResults(actions, req, res, rootSpan, abortController)
 }
 
 function incomingHttpEndTranslator ({ req, res }) {
-  // TODO: this doesn't support headers sent with res.writeHead()
   const responseHeaders = Object.assign({}, res.getHeaders())
   delete responseHeaders['set-cookie']
 
   const payload = {
-    [addresses.HTTP_INCOMING_RESPONSE_CODE]: res.statusCode,
+    [addresses.HTTP_INCOMING_RESPONSE_CODE]: JSON.stringify(res.statusCode),
```

Comment:
why this change?

Reply:
Since we're not coercing all the values to string the status code is passed as a number but the WAF needs it as a string. It is stated in the RFC, if you need more details just let me know.